### PR TITLE
Update FAQ

### PR DESCRIPTION
**Pull request by sergeybykov**

### CODE REVIEW

File: src/Documentation/Frequently-Asked-Questions.md [hunk 1]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
 
 # Frequently Asked Questions
 
-## Does Microsoft support Orleans?
+## Availability
+
+### Can I freely use Orleans in my project?
+
+Absolutely. The source code has been releases under the [MIT license](https://github.com/dotnet/orleans/blob/master/LICENSE). NuGet packages are published on [nuget.org](https://www.nuget.org/profiles/Orleans).
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: typo: released

File: src/Documentation/Frequently-Asked-Questions.md [hunk 2]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
...
+### Microsoft has another actor model implementation - Azure Service Fabric Reliable Actors. How do I choose between the two?
 
-The **runtime behavior perspective** is that an object can only be created within a grain and is not remotely accessible.
-Grains are accessible from anywhere in the system and are location-transparent, so that they can be automatically placed on any server in the deployment, and they survive failures or reboots of servers.
+Reliable Actors are tightly integrated with Service Fabric to leverage its core features, such as replicated in-cluster storage.
+Orleans is not tied to any hosting platform, and can run in almost any environment.
+Orleans provides an [optional integration package](https://www.nuget.org/packages/Microsoft.Orleans.Hosting.ServiceFabric/) for hosting Orleans applications in Service Fabric.
+In the end, it's the application developer's decision of how much they would benefit from the tight integration of Reliable Actors with the underlying platform of Service Fabric versus the flexibility to run anywhere and the feature set of Orleans.
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: We could add a mention about the differences in feature set here, too.

File: src/Documentation/Frequently-Asked-Questions.md [hunk 3]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
...
+### How big or how small should a grain be in my application?
 
-## How should you avoid grain hot spots?
+The grain isolation model makes them very good at representing independent isolated contexts of state and computation.
+In most cases, grains naturally map to such application entities as users, sessions, accounts.
+Those entities are generally isolated from each other, can be accessed and updated independently, and expose a well defined set of supported operations.
+This works well with the intuitive "one entity - one grain" modelling.
+
+An application entity may be too big to be efficiently represented by a single grain if it encapsulates too much state, and as a result has to handle a high rate of requests to it.
+Even a single grain can generally handle up to a few thousand of trivial calls per second, the rule of thumb is to be wary of individual grain receiving hundreds of requests per second.
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: "a few thousand trivial calls"

File: src/Documentation/Frequently-Asked-Questions.md [hunk 4]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
...
+That may be a sign of the grain being too large, and decomposing it into a set of smaller grains may lead to a more stable and balanced system.
+
+An application entity may be too small to be a grain if that would cause constant interaction of other grains with it, and as a result, to too much of messaging overhead.
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: maybe "cause too much of a messaging overhead"?

File: src/Documentation/Frequently-Asked-Questions.md [hunk 5]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
...
+### Can a single Orleans cluster run across multiple datacenters?
+
+Orleans clusters are currently limited to a single data center per cluster.
+Instead, since 1.3.0, you can consider a [multi-cluster deployment where clusters deployed to different datacenters form a single multi-cluster](Multi-Cluster/Overview.md). 
+
+### In what cases can a split brain (same grain activated in multiple silos at the same time) happen?
+
+During normal operations the Orleans runtime guarantees that each grain will have at most one instance in the cluster.
+The only time this guarantee can be violated is when a silo crashes or gets killed without a proper shutdown.
+In that case, there is a ~30 second (based on configuration) window where a grain can potentially get temporarily instantiated in more than one silo.
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: Maybe add "Convergence to a single activation per grain is guaranteed and duplicate activations will be deactivated when this window closes."

File: src/Documentation/Frequently-Asked-Questions.md [hunk 6]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
...
+
+You can find out more about how Orleans manages the clusters at [Cluster Management](Runtime-Implementation-Details/Cluster-Management.md) page.
+
+Also you can take a look at Orleans's [paper](http://research.microsoft.com/pubs/210931/Orleans-MSR-TR-2014-41.pdf) for a more detailed information, however you don't need to understand it fully to be able to write your application code.
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: Orleans' instead of Orleans's

File: src/Documentation/Frequently-Asked-Questions.md [hunk 7]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
...
+You just need to consider the rare possibility of having two instances of an actor while writing your application.
+
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: Should we add that storage providers guarantee that no writes are overwritten without an activation first reading the most up-to-date state?

File: src/Documentation/Frequently-Asked-Questions.md [hunk 8]
```diff
@@ -5,106 +5,128 @@ title: Frequently Asked Questions
...
 
-## Can I persist a grain’s state to the Azure cache service?
-
-This can be done through a storage provider for Azure Cache. We don’t have one but you can easily build your own.
-
-## Can I Connect to Orleans silos from the public internet?
+### Can I Connect to Orleans silos from the public internet?
 
-Orleans is designed to be hosted as the back-end part of a service and you are supposed to create a front-end in your servers which external clients connect to
+Orleans is designed to be hosted as the back-end part of a service, and you expected to create a front-end in your servers which external clients connect to
```
Comment (ReubenBond) [inline src/Documentation/Frequently-Asked-Questions.md]: add . at end

File: src/Documentation/Frequently-Asked-Questions.md (hunk 7 above)
Comment (sergeybykov) [reply src/Documentation/Frequently-Asked-Questions.md]: I phrase it differently. See if you like it.

**Pull request merged**